surrogates/fb-sdk.js

```diff
@@ -0,0 +1,183 @@
+(() => {
+    'use strict';
+    const originalFBURL = document.currentScript.src;
+    let siteInit = function () {};
+    let fbIsEnabled = false;
+    let initData = {};
+    let runInit = false;
+    const parseCalls = [];
+    const popupName = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 12);
+
+    const fbLogin = {
+        callback: function () {},
+        params: undefined,
+        shouldRun: false
+    };
+
+    function messageAddon (detailObject) {
+        detailObject.entity = 'Facebook';
+        const event = new CustomEvent('ddg-ctp', {
+            detail: detailObject,
+            bubbles: false,
+            cancelable: false,
+            composed: false
+        });
+        dispatchEvent(event);
+    }
+
+    /**
+     * When setting up the Facebook SDK, the site may define a function called window.fbAsyncInit.
+     * Once the SDK loads, it searches for and calls window.fbAsyncInit. However, some sites may
+     * not use this, and just call FB.init directly at some point (after ensuring that the script has loaded).
+     *
+     * Our surrogate (defined below in window.FB) captures calls made to init by page scripts. If at a
+     * later point we load the real sdk here, we then re-call init with whatever arguments the page passed in
+     * originally. The runInit param should be true when a page has called init directly.
+     * Because we put it in asyncInit, the flow will be something like:
+     *
+     * FB SDK loads -> SDK calls window.fbAsyncInit -> Our function calls window.FB.init (maybe) ->
+     * our function calls original fbAsyncInit (if it existed)
+     */
+    function enableFacebookSDK () {
+        if (!fbIsEnabled) {
+            window.FB = undefined;
+
+            window.fbAsyncInit = function () {
+                if (runInit && initData) {
+                    window.FB.init(initData);
+                }
+                siteInit();
+                if (fbLogin.shouldRun) {
+                    window.FB.login(fbLogin.callback, fbLogin.params);
+                }
+            };
+
+            const fbScript = document.createElement('script');
+            fbScript.setAttribute('crossorigin', 'anonymous');
+            fbScript.setAttribute('async', '');
+            fbScript.setAttribute('defer', '');
+            fbScript.src = originalFBURL;
+            fbScript.onload = function () {
+                for (const node of parseCalls) {
+                    window.FB.XFBML.parse.apply(window.FB.XFBML, node);
+                }
+            };
+            document.head.appendChild(fbScript);
+            fbIsEnabled = true;
+        } else {
+            if (initData) {
+                window.FB.init(initData);
+            }
+        }
+    }
+
+    function runFacebookLogin () {
+        fbLogin.shouldRun = true;
+        replaceWindowOpen();
+        loginPopup();
+        enableFacebookSDK();
+    }
+
```

Comment:
Just a note to say I've got [a tiny open PR to improve this script](https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/1369). It might be worth waiting for that to land and including that change here, before landing this?